public/javascripts/SVValidate/src/mission/Mission.js

```diff
@@ -103,6 +105,33 @@ function Mission(params) {
         svv.statusField.setProgressText(completionRate);
     }
 
+    /**
+     * Updates status bar (UI) and current mission properties whenever undo is pressed.
+     */
+    function updateMissionProgressUndo() {
+        setProperty("labelsProgress", getProperty("labelsProgress") - 1);
+        setProperty("labelsValidated", getProperty("labelsValidated"));
+        let labelsProgress = getProperty("labelsProgress");
+        let completionRate = labelsProgress / getProperty("labelsValidated");
+        svv.statusField.setProgressBar(completionRate);
+        svv.statusField.setProgressText(completionRate);
+        if (lastStatus === 1) {
+            setProperty('agreeCount', getProperty("agreeCount") - 1);
+        } else if (lastStatus === 2) {
+            setProperty('disagreeCount', getProperty("disagreeCount") - 1);
```

Comment:
`updateValidationResult()` should probably be edited to take in a parameter indicating whether you are adding or removing the validation from the count. That way that function can always take care of updating the counts, and you don't need to basically duplicate the code here.